packages/Kokkos/src/details/DTK_DetailsTreeTraversal.hpp

```diff
@@ -78,26 +23,20 @@ KOKKOS_FUNCTION void
 spatial_query( BVH<NO> const bvh, Predicate const &predicate, int *indices,
                unsigned int &n_indices, unsigned int n_max_indices )
 {
-    // Allocate traversal stack from thread-local memory, and push nullptr to
-    // indicate that there are no postponed nodes.
-    Node const *stack[64];
-    Node const **stack_ptr = stack;
-    *stack_ptr++ = nullptr;
+    Stack<Node const *> stack;
 
     Node const *node = BVHQuery<NO>::getRoot( bvh );
-    unsigned int pos = 0;
-    do
+    stack.push( node );
+    n_indices = 0;
+
+    while ( !stack.empty() )
     {
+        node = stack.top();
+        stack.pop();
+
         if ( BVHQuery<NO>::isLeaf( node ) )
         {
-            indices[pos] = BVHQuery<NO>::getIndex( bvh, node );
-            ++pos;
-#if HAVE_DTK_DBC
-            if ( pos > n_max_indices )
-                printf( "Increase the size of indices array\n" );
-#else
-            (void)n_max_indices;
-#endif
+            indices[n_indices++] = BVHQuery<NO>::getIndex( bvh, node );
```

Comment:
Why did you get ride of the error message if `n_indices` is larger than the size of `indices`.